route/index.js

```diff
@@ -97,21 +97,33 @@ client
  *
  * @param data {object} route specification
  */
-const drawRoutePolyline = data => {
-  const decodedData = mapboxPolyline.decode(data.polyline);
+const drawRoutePolyline = (route, alternatives) => {
+  const routes = [];
+
+  const decodedData = mapboxPolyline.decode(route.polyline);
   const reversed = decodedData.map(item => item.reverse());
 
-  drawRoute(reversed, data.legs);
+  routes.push({ data: route, polyline: reversed });
+
+  alternatives.map(item => {
+    const decoded = mapboxPolyline.decode(item.polyline);
+    const itemReversed = decoded.map(item => item.reverse());
+    routes.push({ data: item, polyline: itemReversed });
+  });
```

Comment:
You are mixing two different styles of array handling: mutating and creating a new array. 

Map always returns a new array but since you are not assigning that result to a variable, forEach would be a better. forEach has 'void' as the return type: it doesn't return anything. 

So either use forEach here or do something like this:


```
const routes = [ { data: route, polyline: reserved }, ...alternatives.map(alternative => ({ data: ..., polyline: .... }) ]